pkg/controller/release.go

```diff
@@ -164,20 +164,29 @@ func (e *helmExternal) Observe(ctx context.Context, mg resource.Managed) (manage
 	}
 
 	cr.Status.AtProvider = generateObservation(rel)
-	u, err := isUpToDate(ctx, e.localKube, &cr.Spec.ForProvider, rel, cr.Status)
+	s, err := isUpToDate(ctx, e.localKube, &cr.Spec.ForProvider, rel, cr.Status)
 	if err != nil {
 		return managed.ExternalObservation{}, errors.Wrap(err, errFailedToCheckIfUpToDate)
 	}
+	cr.Status.Synced = s
+	shouldCallUpdate := !s
+
+	if cr.Status.AtProvider.State == release.StatusDeployed && s {
+		cr.Status.Failed = 0
+	} else if cr.Status.AtProvider.State == release.StatusFailed &&
+		cr.Spec.RollbackLimit != nil && cr.Status.Failed < *cr.Spec.RollbackLimit {
+		shouldCallUpdate = true
+	}
 
 	return managed.ExternalObservation{
 		ResourceExists:   true,
-		ResourceUpToDate: u,
+		ResourceUpToDate: !shouldCallUpdate,
```

Comment:
!cr.status.synced && !rollbackLimitReached would be easier to understand what condition actually triggers the update.